### PR TITLE
Allow variadic template function for secret

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-660

/cc @slackpad

### CODE REVIEW

**Comment by willusher** [comment]
@sethvargo - any reason you're holding off on merging this? Would be much appreciated, thanks.

**Comment by sethvargo** [comment]
I'm waiting on a 👍 from @slackpad 😄

File: template_functions.go [hunk 1]
```diff
@@ -210,15 +210,15 @@ func nodesFunc(brain *Brain,
 
 // secretFunc returns or accumulates secret dependencies from Vault.
 func secretFunc(brain *Brain,
```
Comment (slackpad) [inline template_functions.go]: Do we need to do the same for `secretsFunc()` below? Not sure how that relates to this one.

**Comment by slackpad** [comment]
Gah sorry I missed this. @sethvargo one question on here.

File: template_functions.go (hunk 1 above)
Comment (sethvargo) [inline template_functions.go]: No, `secrets` doesn't currently accept multiple arguments

**Comment by slackpad** [comment]
👍

**Comment by willusher** [comment]
Thank you @sethvargo and @slackpad !

**Pull request merged**